### DISCUSSION TITLE
Hello World Boogie Program for a Custom Programming Language

### DISCUSSION THREAD

**Discussion by lancejpollard** [boogie-org/boogie#542] [Q&A] [answered]
I see in the docs these are "frontend" to boogie:

- [Dafny](https://boogie-docs.readthedocs.io/en/latest/Frontends.html#dafny)
- [Joogie](https://boogie-docs.readthedocs.io/en/latest/Frontends.html#joogie)
- [GPUVerify](https://boogie-docs.readthedocs.io/en/latest/Frontends.html#gpuverify)
- [SMACK](https://boogie-docs.readthedocs.io/en/latest/Frontends.html#smack)
- [VCC](https://boogie-docs.readthedocs.io/en/latest/Frontends.html#vcc)

However, the docs are all currently blank. How would one go about making a new frontend to boogie for a new programming language? I have a programming language in mind, and would like for it to do what dafny does (or boogie), but that's as far as my knowledge goes currently. Not sure where to really begin in connecting a language to boogie, and what the I/O would be to/from the boogie process.

**Comment by shazqadeer** [comment, marked as answer]
See [Forro](https://github.com/boogie-org/forro), a toy language designed to illustrate how to build a verifier using Boogie.
